src/databricks/labs/remorph/reconcile/query_builder_refactored/recon_config.py

```diff
@@ -0,0 +1,132 @@
+    def to_src_col_map(self):
+        if self.column_mapping:
+            return {c.source_name: c.target_name for c in self.column_mapping}
+        return None
+
+    @property
+    def to_tgt_col_map(self):
+        if self.column_mapping:
+            return {c.target_name: c.source_name for c in self.column_mapping}
+        return None
+
+    def get_src_to_tgt_col_mapping(self, cols: list[str] | set[str] | str, layer: str) -> set[str] | str:
+        if layer == "source":
+            return cols
+        if isinstance(cols, list | set):
+            columns = set()
+            for col in cols:
+                columns.add(self.to_src_col_map.get(col, col))
+            return columns
+        return self.to_src_col_map.get(cols, cols)
+
+    def get_tgt_to_src_col_mapping(self, cols: list[str] | set[str] | str, layer: str) -> set[str] | str:
+        if layer == "source":
+            return cols
+        if isinstance(cols, list | set):
+            columns = set()
+            for col in cols:
+                columns.add(self.to_tgt_col_map.get(col, col))
+            return columns
+        return self.to_tgt_col_map.get(cols, cols)
+
+    def get_select_columns(self, schema: list[Schema], layer: str) -> set[str]:
+        if self.select_columns is None:
+            return {sch.column_name for sch in schema}
+        if self.to_src_col_map:
+            return self.get_src_to_tgt_col_mapping(self.select_columns, layer)
+        return set(self.select_columns)
+
+    def get_threshold_columns(self, layer: str) -> set[str]:
+        if self.thresholds is None:
+            return set()
+        return {self.get_src_to_tgt_col_mapping(thresh.column_name, layer) for thresh in self.thresholds}
+
+    def get_join_columns(self, layer: str) -> set[str]:
+        if self.join_columns is None:
+            return set()
+        return {self.get_src_to_tgt_col_mapping(col, layer) for col in self.join_columns}
+
+    def get_drop_columns(self, layer: str) -> set[str]:
+        if self.drop_columns is None:
+            return set()
+        return {self.get_src_to_tgt_col_mapping(col, layer) for col in self.drop_columns}
+
+    def get_transformation_dict(self, layer: str) -> dict[str, str] | None:
+        if self.transformations:
+            if layer == "source":
+                return {t.column_name: t.source for t in self.transformations}
+            return {self.get_src_to_tgt_col_mapping(t.column_name, layer): t.target for t in self.transformations}
+        return None
+
+    def get_partition_column(self, layer: str) -> set[str]:
+        if self.jdbc_reader_options and layer == "source":
+            return {self.jdbc_reader_options.partition_column}
+        return set()
+
+    def get_filter(self, layer: str) -> str | None:
+        if self.filters is None:
+            return None
+        if layer == "source":
+            return self.filters.source
+        return self.filters.target
+
+
+@dataclass
+class Schema:
```

Comment:
this the dataclass need to be used Querybuilder

Reply:
We are using Schema data class until we pass it to Base.py. Inside the Base.py we are converting this to dict for easier operation.